# Evaluating models

Precision: How accurate the recommendations/predictions were

Recall: How complete the recommendations/predictions were

### Classification-based model to evaluate

Is an existing client a potential good candidate for a product?

In [4]:
import pandas as pd
import numpy as np

from pandas import Series, DataFrame
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import classification_report


Data from:
    [Moro et al., 2014] S. Moro, P. Cortez and P. Rita. A Data-Driven Approach to Predict the Success of Bank Telemarketing. Decision Support Systems, Elsevier, 62:22-31, June 2014
         (https://archive.ics.uci.edu/ml/datasets/Bank+Marketing#)

In [6]:
bank = pd.read_csv('bank.csv')
bank.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45211 entries, 0 to 45210
Data columns (total 37 columns):
 #   Column                        Non-Null Count  Dtype 
---  ------                        --------------  ----- 
 0   age                           45211 non-null  int64 
 1   job                           45211 non-null  object
 2   marital                       45211 non-null  object
 3   education                     45211 non-null  object
 4   default                       45211 non-null  object
 5   balance                       45211 non-null  int64 
 6   housing                       45211 non-null  object
 7   loan                          45211 non-null  object
 8   contact                       45211 non-null  object
 9   day                           45211 non-null  int64 
 10  month                         45211 non-null  object
 11  duration                      45211 non-null  int64 
 12  campaign                      45211 non-null  int64 
 13  pdays           

#### We select the relevant columns

In [8]:
x = bank.iloc[:,18:37].values

y = bank.iloc[:,17].values

In [9]:
LogReg = LogisticRegression()
LogReg.fit(x,y)

y_pred = LogReg.predict(x)

#### Evaluating the performance of the model 

In [11]:
print(classification_report(y, y_pred))

              precision    recall  f1-score   support

           0       0.90      0.99      0.94     39922
           1       0.67      0.17      0.27      5289

    accuracy                           0.89     45211
   macro avg       0.79      0.58      0.61     45211
weighted avg       0.87      0.89      0.86     45211



* #### Precision: 87% 
This means 87% of the offers made to customers were liked

* #### Recall: 89% 
This means 89% of all the products customers liked were offered to them